bqplot/pyplot.py

```diff
@@ -816,6 +816,30 @@ def bar(x, y, **kwargs):
     kwargs['y'] = y
     return _draw_mark(Bars, **kwargs)
 
```

Comment:
Why isn't there a _process_data decorator? It will allow the mpl esque dataframe syntax

Reply:
I missed that. But when I was testing it, I think it is better if we do not have it for `boxplot` for now. Because the `y` of `boxplot` is a 2-d array and when we pass a list of column names, the data passed in to the mark is the transpose of what needs to be passed in and hence the plot output is misleading. this is also the case for `Lines`. I feel that is an issue we need to solve separately.

Reply:
Added `process_data` and created issue #548 for fixing the issue mentioned.